### ISSUE TITLE
Mint 17.2 locks up with blank screen after installed TLP

### ISSUE THREAD

**Issue by stevebinns** [linrunner/TLP#151]
After installing TLP on a thinkpad T60 (using the instructions from the TLP web page) the system locks with a blank screen probably after screen save. Sytem info follows then output from tlp-stat
-Computer-
Processor		: 2x Intel(R) Core(TM)2 CPU         T5600  @ 1.83GHz
Memory		: 2063MB (1551MB used)
Operating System		: Linux Mint 17.2 Rafaela
User Name		: steve (Steve Binns)
Date/Time		: Wed 15 Jul 2015 13:51:30 BST
-Display-
Resolution		: 1400x1050 pixels
OpenGL Renderer		: Gallium 0.4 on ATI RV515
X11 Vendor		: The X.Org Foundation
-Multimedia-
Audio Adapter		: HDA-Intel - HDA Intel
Audio Adapter		: ThinkPad EC - ThinkPad Console Audio Control
-Input Devices-
 Lid Switch
 Sleep Button
 Power Button
 AT Translated Set 2 keyboard
 Video Bus
 SynPS/2 Synaptics TouchPad
 Microsoft Basic Optical Mouse
 TPPS/2 IBM TrackPoint
 ThinkPad Extra Buttons
-Printers (CUPS)-
Canon-MG5300		: <i>Default</i>
Canon-MG6400
NewCanon
PDF
-SCSI Disks-
HL-DT-ST DVDRAM GSA-4083N
ATA HITACHI HTS54161
--------------------------------- tlp-stat
--- TLP 0.7 --------------------------------------------

+++ Configured Settings: /etc/default/tlp
TLP_ENABLE=1
DISK_IDLE_SECS_ON_AC=0
DISK_IDLE_SECS_ON_BAT=2
MAX_LOST_WORK_SECS_ON_AC=15
MAX_LOST_WORK_SECS_ON_BAT=60
SCHED_POWERSAVE_ON_AC=0
SCHED_POWERSAVE_ON_BAT=1
NMI_WATCHDOG=0
ENERGY_PERF_POLICY_ON_AC=performance
ENERGY_PERF_POLICY_ON_BAT=powersave
DISK_DEVICES="sda sdb"
DISK_APM_LEVEL_ON_AC="254 254"
DISK_APM_LEVEL_ON_BAT="128 128"
SATA_LINKPWR_ON_AC=max_performance
SATA_LINKPWR_ON_BAT=min_power
PCIE_ASPM_ON_AC=performance
PCIE_ASPM_ON_BAT=powersave
RADEON_POWER_PROFILE_ON_AC=high
RADEON_POWER_PROFILE_ON_BAT=low
RADEON_DPM_STATE_ON_AC=performance
RADEON_DPM_STATE_ON_BAT=battery
RADEON_DPM_PERF_LEVEL_ON_AC=auto
RADEON_DPM_PERF_LEVEL_ON_BAT=auto
WIFI_PWR_ON_AC=1
WIFI_PWR_ON_BAT=5
WOL_DISABLE=Y
SOUND_POWER_SAVE_ON_AC=0
SOUND_POWER_SAVE_ON_BAT=1
SOUND_POWER_SAVE_CONTROLLER=Y
BAY_POWEROFF_ON_BAT=0
BAY_DEVICE="sr0"
RUNTIME_PM_ON_AC=on
RUNTIME_PM_ON_BAT=auto
RUNTIME_PM_ALL=1
RUNTIME_PM_DRIVER_BLACKLIST="radeon nouveau"
USB_AUTOSUSPEND=1
USB_DRIVER_BLACKLIST="usbhid"
USB_BLACKLIST_WWAN=1
RESTORE_DEVICE_STATE_ON_STARTUP=0

+++ System Info
System         = LENOVO ThinkPad T60 2007FVG
BIOS           = 79ETD2WW (2.12 )
Release        = Linux Mint 17.2 Rafaela
Kernel         = 3.13.0-24-generic i686
/proc/cmdline  = BOOT_IMAGE=/boot/vmlinuz-3.13.0-24-generic root=UUID=7a4e83c0-e306-4ff2-b3d7-c1412ab62aa6 ro quiet splash vt.handoff=7
Init system    = upstart

+++ System Status
TLP power save = enabled
power source   = ac

+++ Processor
CPU Model      = Intel(R) Core(TM)2 CPU         T5600  @ 1.83GHz

/sys/devices/system/cpu/cpu0/cpufreq/scaling_driver    = acpi-cpufreq
/sys/devices/system/cpu/cpu0/cpufreq/scaling_governor  = ondemand
/sys/devices/system/cpu/cpu0/cpufreq/scaling_min_freq  =  1000000 [kHz]
/sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq  =  1833000 [kHz]
/sys/devices/system/cpu/cpu0/cpufreq/scaling_available_frequencies = 1833000 1333000 1000000 [kHz]

/sys/devices/system/cpu/cpu1/cpufreq/scaling_driver    = acpi-cpufreq
/sys/devices/system/cpu/cpu1/cpufreq/scaling_governor  = ondemand
/sys/devices/system/cpu/cpu1/cpufreq/scaling_min_freq  =  1000000 [kHz]
/sys/devices/system/cpu/cpu1/cpufreq/scaling_max_freq  =  1833000 [kHz]
/sys/devices/system/cpu/cpu1/cpufreq/scaling_available_frequencies = 1833000 1333000 1000000 [kHz]

/sys/devices/system/cpu/cpufreq/boost                  = 0 (cpu not supported)

x86_energy_perf_policy: program not installed.

/proc/sys/kernel/nmi_watchdog                          = 0

+++ Undervolting
PHC kernel not available.

+++ Temperatures
/proc/acpi/ibm/thermal = 53 37 35 56 -128 -128 -128 -128 37 49 52 -128 -128 -128 -128 -128 [°C]
/proc/acpi/ibm/fan     =  3460 [/min]

+++ File System
/proc/sys/vm/laptop_mode               =     0
/proc/sys/vm/dirty_writeback_centisecs =  1500
/proc/sys/vm/dirty_expire_centisecs    =  1500
/proc/sys/vm/dirty_ratio               =    20
/proc/sys/vm/dirty_background_ratio    =    10
/proc/sys/fs/xfs/age_buffer_centisecs  = (not available)
/proc/sys/fs/xfs/xfssyncd_centisecs    = (not available)
/proc/sys/fs/xfs/xfsbufd_centisecs     = (not available)

+++ Storage Devices
/dev/sda:
          Model     = HITACHI HTS541612J9SA00                 
          Firmware  = SBDIC7EP
          APM Level = 254
          Status    = active/idle
          Scheduler = deadline

        SMART info:
            4 Start_Stop_Count          =     2289 
            5 Reallocated_Sector_Ct     =        0 
            9 Power_On_Hours            =     6157 [h]
          193 Load_Cycle_Count          =    35480 
          194 Temperature_Celsius       =       33 (Min/Max 5/51)  [°C]

+++ SATA Aggressive Link Power Management
/sys/class/scsi_host/host2/link_power_management_policy  = max_performance
/sys/class/scsi_host/host3/link_power_management_policy  = max_performance
/sys/class/scsi_host/host4/link_power_management_policy  = max_performance
/sys/class/scsi_host/host5/link_power_management_policy  = max_performance

+++ PCIe Active State Power Management
/sys/module/pcie_aspm/parameters/policy = default (using bios preferences)

+++ Radeon Graphics
/sys/class/drm/card0/device/power_method = profile
/sys/class/drm/card0/device/power_profile = high

+++ Wireless
bluetooth = on
wifi      = on
wwan      = none (no device)

hci0(btusb)         : not connected
wlan0(ath9k)        : connected, power management = off

+++ Audio
/sys/module/snd_hda_intel/parameters/power_save            = 0
/sys/module/snd_hda_intel/parameters/power_save_controller = N

+++ Docks and Device Bays
/sys/devices/platform/dock.0: battery_bay   = no battery 
/sys/devices/platform/dock.1: ata_bay       = drive present
/sys/devices/platform/dock.2: dock_station  = undocked

+++ Runtime Power Management
device classes   = all
device blacklist = (not configured)
driver blacklist = radeon nouveau

/sys/bus/pci/devices/0000:00:00.0/power/control = on   (0x060000, Host bridge, no driver)
/sys/bus/pci/devices/0000:00:01.0/power/control = on   (0x060400, PCI bridge, pcieport)
/sys/bus/pci/devices/0000:00:1b.0/power/control = on   (0x040300, Audio device, snd_hda_intel)
/sys/bus/pci/devices/0000:00:1c.0/power/control = on   (0x060400, PCI bridge, pcieport)
/sys/bus/pci/devices/0000:00:1c.1/power/control = on   (0x060400, PCI bridge, pcieport)
/sys/bus/pci/devices/0000:00:1c.2/power/control = on   (0x060400, PCI bridge, pcieport)
/sys/bus/pci/devices/0000:00:1c.3/power/control = on   (0x060400, PCI bridge, pcieport)
/sys/bus/pci/devices/0000:00:1d.0/power/control = on   (0x0c0300, USB controller, uhci_hcd)
/sys/bus/pci/devices/0000:00:1d.1/power/control = on   (0x0c0300, USB controller, uhci_hcd)
/sys/bus/pci/devices/0000:00:1d.2/power/control = on   (0x0c0300, USB controller, uhci_hcd)
/sys/bus/pci/devices/0000:00:1d.3/power/control = on   (0x0c0300, USB controller, uhci_hcd)
/sys/bus/pci/devices/0000:00:1d.7/power/control = on   (0x0c0320, USB controller, ehci-pci)
/sys/bus/pci/devices/0000:00:1e.0/power/control = on   (0x060401, PCI bridge, no driver)
/sys/bus/pci/devices/0000:00:1f.0/power/control = on   (0x060100, ISA bridge, lpc_ich)
/sys/bus/pci/devices/0000:00:1f.1/power/control = on   (0x01018a, IDE interface, ata_piix)
/sys/bus/pci/devices/0000:00:1f.2/power/control = on   (0x010601, SATA controller, ahci)
/sys/bus/pci/devices/0000:00:1f.3/power/control = on   (0x0c0500, SMBus, no driver)
/sys/bus/pci/devices/0000:01:00.0/power/control = on   (0x030000, VGA compatible controller, radeon)
/sys/bus/pci/devices/0000:02:00.0/power/control = on   (0x020000, Ethernet controller, e1000e)
/sys/bus/pci/devices/0000:03:00.0/power/control = on   (0x028000, Network controller, ath9k)
/sys/bus/pci/devices/0000:15:00.0/power/control = on   (0x060700, CardBus bridge, yenta_cardbus)

+++ USB
autosuspend        = enabled
device whitelist   = (not configured)
device blacklist   = (not configured)
driver blacklist   = usbhid
wwan blacklist     = enabled (cdc_acm cdc_wdm cdc_ether hso qcserial sierra)

Bus 001 Device 001 ID 1d6b:0002 control = auto, autosuspend_delay_ms =     0 -- Linux Foundation 2.0 root hub (hub)
Bus 005 Device 003 ID 0483:2016 control = auto, autosuspend_delay_ms =  2000 -- STMicroelectronics Fingerprint Reader (no driver)
Bus 005 Device 002 ID 0a5c:2110 control = auto, autosuspend_delay_ms =  2000 -- Broadcom Corp. BCM2045B (BDC-2) [Bluetooth Controller] (btusb)
Bus 005 Device 001 ID 1d6b:0001 control = auto, autosuspend_delay_ms =     0 -- Linux Foundation 1.1 root hub (hub)
Bus 004 Device 001 ID 1d6b:0001 control = auto, autosuspend_delay_ms =     0 -- Linux Foundation 1.1 root hub (hub)
Bus 003 Device 001 ID 1d6b:0001 control = auto, autosuspend_delay_ms =     0 -- Linux Foundation 1.1 root hub (hub)
Bus 002 Device 002 ID 045e:0083 control = on,   autosuspend_delay_ms =  2000 -- Microsoft Corp. Basic Optical Mouse (usbhid)
Bus 002 Device 001 ID 1d6b:0001 control = auto, autosuspend_delay_ms =     0 -- Linux Foundation 1.1 root hub (hub)

+++ ThinkPad Extended Battery Functions
tp-smapi   = active
tpacpi-bat = inactive (unsupported hardware)

+++ Suggestions
* Install ethtool to disable Wake On LAN (WOL)

**Comment by linrunner** [comment]
Hi, please clarify your description:

> the system locks with a blank screen probably(?) after screen save

What are you doing and what happens exactly?

I suggest to uninstall TLP and check if the symptom disappears.

**Comment by stevebinns** [comment]
Unfortunately it's not particularly reproducible. I did remove TLP and the system behaved OK. I've now re-installed TLP and seen the following.
1 System seems OK with battery removed, with external power. 
2 With battery installed and no external power, starting a lock screen produces a blank screen (not the usual flying star field) and moving the mouse gives a very jerky movement of the cursor and the password request does not appear. Putting external power on fixes this problem.
3 I then left the system with battery and charger operating and came back to aforementioned black screen with no way to exit except turn of power and remove battery.
4 After this tried to reboot with just battery installed and got background screen with no icons or aplets and no response to mouse or keyboard.

Sorry it's all a bit vague. 
2 With battery installed and no external power, starting a lock screen produces a blank screen (not the usual flying star field) and moving the mouse gives a very jerky movement of the cursor and the password request does not appear. Putting external power on fixes this problem.
3 I then left the system with battery and charger operating and came back to aforementioned black screen with no way to exit except turn of power and remove battery.
4 After this tried to reboot with just battery installed and got background screen with no icons or aplets and no response to mouse or keyboard.

Sorry it's all a bit vague.

**Comment by linrunner** [comment]
Your BIOS version is *way* outdated. Update to the [latest version](http://support.lenovo.com/us/en/products/laptops-and-netbooks/thinkpad-t-series-laptops/thinkpad-t60/downloads/DS013722).

**Comment by stevebinns** [comment]
OK thanks, I've now done that. 

Behaviour a bit better but a long way from perfect. Typical behaviour running on battery:
start up system, start chrome browser - takes about 1 min. Close chrome and restart it, it creates an icon on the panel but won't display the window. If I click on the icon it's as if I double click. It clears the desktop image then quickly redraws it. The chrome window never appears. With battery and charger chrome starts up in about 20 odd secs and operated normally - it's what I'm using now. I'll remove TLP and try again

**Comment by stevebinns** [comment]
Now removed TLP and running on battery, chrome starts up in less than 20 secs and can be stopped and restarted with no apparent problem. Started lock screen with no problem. Anything else to try?

**Comment by linrunner** [comment]
I can only guess at this stage:

(1) You should get the latest updates for kernel 3.13, as Mint (sadly) defaults to not update kernels:

    sudo apt-get install linux-generic

or your may try 3.16 with

    sudo apt-get install linux-generic-lts-utopic

(2) Related to your performance issue i would experiment with less aggressive values for DISK_APM_LEVEL_ON_BAT and SATA_LINKPWR_ON_BAT – see [TLP settings docs](http://linrunner.de/en/tlp/docs/tlp-configuration.html#disks) 

(3) Regarding graphics i would experiment with
a. comment out RUNTIME_PM_ON_* 
b. choose less aggressive radeon settings or try to enable DPM – see [docs](http://linrunner.de/en/tlp/docs/tlp-configuration.html#graphics)

Remark: the ATI X1400 is nice especially in winter (*huge* savings on your heating bill) and the radeon driver is particularly buggy for that old chip. I had a similarly specced T60 some years ago. Maybe this is the root of your problems with power saving.

I happen to have installed an Intel Graphics T60 (Hitachi HDD) with Xubuntu 14.04.2 this morning and observed none of your symptoms.

**Comment by stevebinns** [comment]
Many thanks for this info. I'll pursue a bit further but as you say the problems may well be all down to the graphics set - the system does get very hot. On a separate issue (as you seem very knowledgeable!)  I've noticed that I can't get the screen brightness on battery power up to the same level as when on external power. Can this be fixed?

**Comment by linrunner** [comment]
In the BIOS: Config --> Display --> Brightness

**Comment by stevebinns** [comment]
OK many thanks for your help. I've upgraded the kernel to 3.13 and enabled DPM. I still have the freeze problem with TLP on battery. Typically I have a chrome window I can move the cursor around but clicking anywhere has no effect. I have to hold down the power button for 5 secs for it to halt. I can't see that tweaking settings would alter that behaviour so I've removed TLP for now. The screen is brighter now too. Thanks again.
